src/iuse.cpp

```diff
@@ -2339,21 +2169,30 @@ int iuse::radio_mod( player *p, item *, bool, const tripoint& )
 
 int iuse::remove_all_mods(player *p, item *, bool, const tripoint& )
 {
-    static const std::vector<std::string> removable_mods = {{
-        "DOUBLE_AMMO", "USE_UPS", "ATOMIC_AMMO"
-    }};
+    if( !p ) {
```

Comment:
`Character &user, item &it, bool t, const tripoint &pos`

If using `item_location`, you should first guarantee that the supplied location:
* Always points to a valid item (this should never need to be checked in the iuses)
* Can be accessed in constant time almost always (ie. unpacking would be done only for the first time)
* `remove_item` is never invoked during processing. Otherwise the active item vectors would get scrambled and the function itself could segfault.

So overall it would be better to keep it without location. If we want location, it would be better to keep an item map and give items some sort of id - it would be more work at start, but would most likely end up less work in total.